contributed_definitions/nyaml/NXpiezoelectric_material.yaml

```diff
@@ -0,0 +1,191 @@
+category: base
+doc: |
+  Description and properties of the piezoelectric actuator materials.
+  The piezoelectric actuator is usually composed of polycrystalline solids and
+  attached at the head of the tip or cantilever. The material deforms when the external electric field is applied.
+type: group
+NXpiezoelectric_material(NXobject):
+  
+  # Online links:
+  # 1. ttps://www.ulprospector.com/knowledge/2689/pe-piezoelectric-materials/#:~:text=piezoelectric ceramics, such as lithium,as polyvinylidene fluoride (PVDF).
+  # 2. https://info.piezo.com/hubfs/Data-Sheets/piezo-material-properties-data-sheet-20201112.pdf
+  # 3. https://application.wiley-vch.de/books/sample/3527345124_c01.pdf
+  name(NX_CHAR):
+    doc: |
+      The name of the material of the piezo scanner such as Lead Zirconate Titanates
+      (PZTs).
+  chemical_description(NXsubstance):
+    doc: |
+      The chemical formula of the material of the piezo scanner such as Pb(Zr,Ti)O3.
+  type(NX_CHAR):
+    doc: |
+      The type of the material of the piezo scanner (e.g. piezoelectric ceramics,
+      polymer-film piezoelectrics).
+  
+  # Properties of the piezo material
+  density(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The density of the piezo material.
+  relative_permittivity(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      The relative permittivity (dielectric constant) of the piezo material.
+  D_piezoelectric_constant(NX_NUMBER):
```

Comment:
you need to set `nameType=partial` to make the `D` part renameable. Also below.

Reply:
See below.